hometask2/hometask2_1.js

```diff
@@ -0,0 +1,145 @@
+const express = require("express");
+const app = express();
+
+const bodyParser = require("body-parser");
+app.use(bodyParser.json() );
+app.use(bodyParser.urlencoded({
+  extended: true
+})); 
+
+
+const Joi = require("@hapi/joi");
+const validator = require("express-joi-validation").createValidator({});
+
+const schema = Joi.object({
+	login: Joi.string()
+		.alphanum()
+		.min(5)
+		.max(30)
+		.required(),
+
+	password: Joi.string()
+		.pattern(new RegExp("^[a-zA-Z0-9]{8,30}$"))
+		.required(),
+
+	age: Joi.number()
+		.integer()
+		.min(4)
+		.max(130)
+		.required(),
+});
+
+
+
+const usersMap = new Map();
+let uid = 1;
+
+//read
+app.get("/user/:id", (req, res) => {
+	
+	let id = req.params.id;
+	
+	let user = usersMap.get(id);
+	if (user && !user.isDeleted) {
+		res.send("user = " + JSON.stringify(user));
+	} else {
+		res.send("user not found, id = " + id);
+	}
+	
+});
+
+//autosuggest
+app.get("/users/:login/:limit", (req, res) => {
+	
+	const login = req.params.login;
+	let limit = parseInt(req.params.limit);
+
+	let count = 0;
+	let userResult = new Array();
```

Comment:
"new Array()" is a pure Java way of creating an array in JS and it is frowned upon for its behaviour.
Consider using array literal "[]" instead.